src/day3.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <vector>
+#include <fstream>
+
+std::vector<std::string> readPuzzleInputFromFile(std::string fileName) {
+    std::ifstream inFile(fileName);
+    std::string line;
+    std::vector<std::string> linesInFile{};
+
+    while (getline(inFile, line)) {
+        linesInFile.push_back(line);
+    }
+    return linesInFile;
+}
+
+int numberOfTreesOnSlope(const std::vector<std::string>& area, std::vector<int> slope){
+    int index = 0;
+    int counter = 0;
+    int stepRight = slope[0];
+    int stepDown = slope[1];
+    for (std::vector<std::string>::const_iterator it = area.begin(); it != area.end() + stepDown - 1; it += stepDown) {
+            if ((*it)[index % (*it).length()] == '#') {
+                counter++;
+            }
+            index += stepRight;
+        }
+    return counter;
+}
+
+int productOfTreesOnSlopes(const std::vector<std::string>& inputLines, const std::vector<std::vector<int>>& slopes) {
+    
+    std::vector<int> numberOfTreesPerSlope{};
+
+    for (std::vector<std::vector<int>>::const_iterator it = slopes.begin(); it != slopes.end(); it++) {
+        int numberOfTrees = numberOfTreesOnSlope(inputLines, *it);
+        numberOfTreesPerSlope.push_back(numberOfTrees);
+    } 
+
+    int product = 1;
+    for (auto& numberOfTrees : numberOfTreesPerSlope){
+        product *= numberOfTrees;
```

Comment:
It's actually a shame you're computing the product here. If you would compute the product directly in the loop in which you loop over `slopes`, you wouldn't need `numberOfTreesPerSlope`!